service/topology_coordinator.cc

```diff
@@ -1802,17 +1840,21 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                         co_await _group0.make_nonvoter(replaced_node_id, _as);
                     }
                 }
+
                 utils::get_local_injector().inject("crash_coordinator_before_stream", [] { abort(); });
                 raft_topology_cmd cmd{raft_topology_cmd::command::stream_ranges};
                 auto state = node.rs->state;
                 try {
+                    // No need to stream data when the node is zero-token.
+                    if (!node.rs->ring.tokens.empty()) {
```

Comment:
Didn't we move such nodes straight to normal above. They do not suppose to go through streaming stage, no?

Reply:
Move move joining zero-token node straight to normal. This code applies to removing and decommissioning zero-token nodes.